util/src/io/worker.rs

```diff
@@ -123,7 +123,7 @@ impl Worker {
 impl Drop for Worker {
 	fn drop(&mut self) {
 		trace!(target: "shutdown", "[IoWorker] Closing...");
-		let _ = self.wait_mutex.lock();
+		let _ = self.wait_mutex.lock().unwrap();
 		self.deleting.store(true, AtomicOrdering::Release);
 		self.wait.notify_all();
```

Comment:
I wonder if the issue is that this call happens before the atomic store due to compiler or CPU isntruction reordering.
System condvars can have spurious wakeups, but `parking_lot` ones can't. I don't have a windows machine right now, but could you test using `parking_lot` and adding a call to `atomic::fence(AtomicOrdering::SeqCst)` between the store to `deleting` and the condvar notification?